### PR TITLE
K8SPS-17: Integrate pt-heartbeat

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPS-17](https://jira.percona.com/browse/K8SPS-17) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/32d3f3afb8c6e257085b076e1afb514631ddc2ae
image: `perconalab/percona-server-mysql-operator:PR-164-32d3f3a`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/18077028011aba3be887104d67e95a12a800ca70
image: `perconalab/percona-server-mysql-operator:PR-164-1807702`

**Review by inelpandzic — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/auto-config.log) |
| config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/demand-backup.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/init-deploy.log) |
| gr-init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/gr-init-deploy.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/scaling.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/users.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/monitoring.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/service-per-pod.log) |
| sidecars | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/sidecars.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/d2cecc8/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/d2cecc8871188cdb3bdee2c05edcd78f2c4a4d70
image: `perconalab/percona-server-mysql-operator:PR-164-d2cecc8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| We run 0 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/0207250d0cb8b28ee568a9871e3c3412d6c9acbc
image: `perconalab/percona-server-mysql-operator:PR-164-0207250`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/demand-backup.log) |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/init-deploy.log) |
| config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/config.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/scaling.log) |
| gr-init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/gr-init-deploy.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/users.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/monitoring.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/service-per-pod.log) |
| sidecars | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/sidecars.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/1076302/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/1076302125ad5f3e0cee761f1a5ab0a6afaeeff3
image: `perconalab/percona-server-mysql-operator:PR-164-1076302`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/init-deploy.log) |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/demand-backup.log) |
| gr-init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/gr-init-deploy.log) |
| config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/config.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/scaling.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/monitoring.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/users.log) |
| semi-sync | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/semi-sync.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/service-per-pod.log) |
| sidecars | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/sidecars.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/0608b6b/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/0608b6b7f9deb6f5d8141f89836f31e567cd81cb
image: `perconalab/percona-server-mysql-operator:PR-164-0608b6b`

File: build/ps-entrypoint.sh [hunk 1]
```diff
@@ -321,7 +322,12 @@ if [ "$1" = 'mysqld' -a -z "$wantHelp" ]; then
 			CREATE USER 'orchestrator'@'%' IDENTIFIED BY '${ORC_TOPOLOGY_PASSWORD}';
 			GRANT SYSTEM_USER, SUPER, PROCESS, REPLICATION SLAVE, REPLICATION CLIENT, RELOAD ON *.* TO 'orchestrator'@'%';
 			GRANT SELECT ON mysql.slave_master_info TO 'orchestrator'@'%';
-			GRANT SELECT ON meta.* TO 'orchestrator'@'%';
+			GRANT SELECT ON sys_operator.* TO 'orchestrator'@'%';
+
+			CREATE DATABASE sys_operator;
```
Comment (hors) [inline build/ps-entrypoint.sh:327]:
```suggestion
			CREATE DATABASE IF NOT EXISTS sys_operator;
```

File: deploy/cr.yaml [hunk 2]
```diff
@@ -44,7 +47,7 @@ spec:
       persistentVolumeClaim:
         resources:
           requests:
-            storage: 2G
+            storage: 20G
```
Comment (hors) [inline deploy/cr.yaml:50]: Why do we need 20G? Can we use 6G as in PXCC operator?

File: deploy/cr.yaml [hunk 3]
```diff
@@ -7,6 +7,9 @@ spec:
 #  pause: false
   secretsName: cluster1-secrets
   sslSecretName: cluster1-ssl
+  toolkit:
```
Comment (hors) [inline deploy/cr.yaml:10]: Did you ask @spron-in about this option/section ? We need to confirm all UI changes with him.

**Review by hors — CHANGES_REQUESTED** [review]

File: build/ps-entrypoint.sh (hunk 1 above)
Comment (egegunes) [reply build/ps-entrypoint.sh]: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/36104b4e6d61f6adb42251c38cf7efc56824cdca

File: deploy/cr.yaml (hunk 2 above)
Comment (egegunes) [reply deploy/cr.yaml]: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/f1c37b264d833ef3d9ba3f299d7051f721825f97

File: deploy/cr.yaml (hunk 3 above)
Comment (egegunes) [reply deploy/cr.yaml:10]: @spron-in ping

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/gr-init-deploy.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/monitoring.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/scaling.log) |
| config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/config.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/users.log) |
| sidecars | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/sidecars.log) |
| limits | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/276043f/limits.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/276043f94e681e52cfd6bfd831ed6c7527e02787
image: `perconalab/percona-server-mysql-operator:PR-164-276043f`

File: deploy/cr.yaml (hunk 3 above)
Comment (spron-in) [reply deploy/cr.yaml:10]: name looks ok to me.

I would move toolkit section to the bottom of the default cr.yaml and add commented resources subsection:
```
resources:
  requests:
    cpu:
    memory: 
  limits:
    cpu:
    memory:
```

@fiowro we will need to add info about toolkit usage to our docs. Architecture and options for sure.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/gr-init-deploy.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/scaling.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/semi-sync.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/users.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/limits.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/89d0221/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/89d022107ac5af3b1fee4e6b16f486310b064f30
image: `perconalab/percona-server-mysql-operator:PR-164-89d0221`

File: build/heartbeat-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+set -o xtrace
+
+until [ ! -f /var/lib/mysql/bootstrap.lock ] && [ ! -f /var/lib/mysql/clone.lock ]; do
+	echo "waiting for MySQL initialization"
+	sleep 10
+done
+
+sleep 60
```
Comment (hors) [inline build/heartbeat-entrypoint.sh:10]: I am not sure about this 'sleep 60'. Why do we need it at all?

File: build/heartbeat-entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+pt-heartbeat \
+	--update \
+	--replace \
+	--fail-successive-errors 20 \
+	--check-read-only \
+	--create-table \
+	--database sys_operator \
+	--table heartbeat \
+	--user heartbeat \
+	--password "${HEARTBEAT_PASSWORD}" \
```
Comment (hors) [inline build/heartbeat-entrypoint.sh:22]: we will print password in the log

**Review by hors — CHANGES_REQUESTED** [review]

File: build/heartbeat-entrypoint.sh (hunk 5 above)
Comment (hors) [reply build/heartbeat-entrypoint.sh:22]: Why  do we use env var? If I want to change this  password mysql pod will be restarted :(

File: api/v1alpha1/perconaservermysql_types.go [hunk 6]
```diff
@@ -102,39 +103,37 @@ type OrchestratorSpec struct {
 	PodSpec `json:",inline"`
 }
 
+type ContainerSpec struct {
+	Image            string                        `json:"image,omitempty"`
+	ImagePullPolicy  corev1.PullPolicy             `json:"imagePullPolicy,omitempty"`
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	Resources        corev1.ResourceRequirements   `json:"resources,omitempty"`
```
Comment (hors) [inline api/v1alpha1/perconaservermysql_types.go:110]: I have tried to changed resources but the pods were not restarted

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/init-deploy.log) |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/auto-config.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/demand-backup.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/monitoring.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/scaling.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/semi-sync.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/users.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/limits.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/4369a56/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/4369a568acc591c8f2177294dde327f3b12cd16c
image: `perconalab/percona-server-mysql-operator:PR-164-4369a56`

File: pkg/mysql/mysql.go [hunk 7]
```diff
@@ -557,6 +564,40 @@ func backupContainer(cr *apiv1alpha1.PerconaServerMySQL) corev1.Container {
 	}
 }
 
+func heartbeatContainer(cr *apiv1alpha1.PerconaServerMySQL) corev1.Container {
+	return corev1.Container{
+		Name:            "pt-heartbeat",
+		Image:           cr.Spec.Toolkit.Image,
+		ImagePullPolicy: cr.Spec.Toolkit.ImagePullPolicy,
```
[the remark this answers is not in the document]
Comment (hors) [reply pkg/mysql/mysql.go:571]: [28db6c4](https://github.com/percona/percona-server-mysql-operator/pull/164/commits/28db6c4edc609d03da804016d4f32180be06ff14)

**Review by hors — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/users.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/b917a34/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/b917a348efc5ed254f2e895f8789964181ac3204
image: `perconalab/percona-server-mysql-operator:PR-164-b917a34`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/semi-sync.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/scaling.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/service-per-pod.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/users.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/limits.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/5f770c4/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/5f770c4fcdc60d9f425603ccf85337c7a183f5e3
image: `perconalab/percona-server-mysql-operator:PR-164-5f770c4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/users.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/7121b4175717e3a6cae534dab21e16c828fcbd03
image: `perconalab/percona-server-mysql-operator:PR-164-7121b41`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/semi-sync.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/service-per-pod.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/scaling.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/users.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/7121b41/config.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/7121b4175717e3a6cae534dab21e16c828fcbd03
image: `perconalab/percona-server-mysql-operator:PR-164-7121b41`

**Review by inelpandzic — APPROVED** [review]

File: build/heartbeat-entrypoint.sh [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+# we can get situation when ps-entrypoint removed bootstrap.lock but bootstrap has not created clone.lock yet
+sleep 10
+if [ -f /var/lib/mysql/clone.lock ]; then
+	CLONE_IN_PROGRESS='yes'
+fi
+
+MYSQL_ADMIN_PORT='33062'
+MYSQL_USER="${MYSQL_USERNAME:-monitor}"
+MONITOR_PASSWOR=$(cat /etc/mysql/mysql-users-secret/monitor || :)
+MYSQL_PASSWORD="${MONITOR_PASSWOR:-$MONITOR_PASSWORD}"
```
[the remark this answers is not in the document]
Comment (hors) [reply build/heartbeat-entrypoint.sh]: fixed

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| auto-config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/auto-config.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/init-deploy.log) |
| gr-init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/gr-init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/monitoring.log) |
| semi-sync | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/semi-sync.log) |
| config | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/config.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/service-per-pod.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/scaling.log) |
| sidecars | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/sidecars.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-ps-operator/PR-164/61c2877/users.log) |
| We run 12 out of 12|

commit: https://github.com/percona/percona-server-mysql-operator/pull/164/commits/61c287759140c7b013f37488a2f5739aa8ff9f97
image: `perconalab/percona-server-mysql-operator:PR-164-61c2877`

**Review by pooknull — APPROVED** [review]

**Pull request merged**